### ISSUE TITLE
How many columns can the parseCSV handle?

### ISSUE THREAD

**Issue by Rob-ModernAnalytics** [data-forge/data-forge-ts#58]
Hi,

I'm trying to read in a csv file that has about 4500 columns.  On the .parseCSV method I am getting this error (I've parsed other CSV files I have and it works fine - suspecting there may be an issue with the number of columns??):

testDataForge Error Error: "toString()" failed
    at stringSlice (buffer.js:558:43)
    at Buffer.toString (buffer.js:631:10)
    at Object.fs.readFileSync (fs.js:601:41)
    at SyncFileReader.parseCSV (D:\Applications\ModernAnalytics\TestDataForge\node_modules\data-forge-fs\build\index.js:324:40)
    at __dirname (D:\Applications\ModernAnalytics\TestDataForge\testDataForge.js:7:76)
    at Object.<anonymous> (D:\Applications\ModernAnalytics\TestDataForge\testDataForge.js:26:3)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)

Thanks!

Rob

**Comment by Rob-ModernAnalytics** [comment]
Hi,

After a bit more experimenting I tried loading a much smaller file with the same number of columns and it worked fine.  The file has about 15K records in it.  I'll work more on the file to see where it starts failing.

Thanks,

Rob

**Comment by ashleydavis** [comment]
There is a maximum string length in Node.js and it might be failing when it hits that

I've got a blog post that you can read about these limits and how to increase them: http://www.the-data-wrangler.com/nodejs-memory-limits/

Let me know how it goes!

**Comment by Rob-ModernAnalytics** [comment]
Hi Ashley,

I experimented more with reading my csv with data-forge:

  1.  I had to increase the max-old-space-size but that really didn’t solve the problem.
  2.  I am able to read 13,725 records of my file (including column header record).  When I try one more record it fails.  It really doesn’t matter which record (I deleted various records) – so it’s not something a particular record.
  3.  The strange part is that my test program crashes immediately when it hits the .parseCSV line of code when my file is > 13,725 records.  It’s not like a memory problem where the program runs and runs then crashes.

Thanks,

Rob
